src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtils.java

```diff
@@ -425,4 +427,33 @@ public static boolean isReceiverParameter(DetailAST parameterDefAst) {
         }
         return returnValue;
     }
+
+    /**
+     * Returns {@link AccessModifier} based on the information about access modifier
+     * taken from the given token of type {@link TokenTypes#MODIFIERS}.
+     * @param modifiersToken token of type {@link TokenTypes#MODIFIERS}.
+     * @return {@link AccessModifier}.
+     */
+    public static AccessModifier getAccessModifierFromModifiersToken(DetailAST modifiersToken) {
+        if (modifiersToken == null || modifiersToken.getType() != TokenTypes.MODIFIERS) {
+            throw new IllegalArgumentException("expected non-null AST-token with type 'MODIFIERS'");
+        }
+
+        // default access modifier
+        AccessModifier accessModifier = AccessModifier.PACKAGE;
+        for (AST token = modifiersToken.getFirstChild(); token != null;
+             token = token.getNextSibling()) {
+
+            if (AccessModifier.PUBLIC.getName().equals(token.getText())) {
```

Comment:
I think you mis-understood what I meant. I meant token type like,
`if (TokenTypes.LITERAL_PUBLIC == token.getType()) {`
I see no reason to compare any type of text, when these tokens have standard integer types that already specify them.